### PR TITLE
[CLEANUP beta] Remove bind-attr

### PR DESCRIPTION
**Pull request by mmun**

### CODE REVIEW

**Comment by mmun** [comment]
We should extract this to an addon before merging. Anyone want to volunteer?

**Comment by mmun** [comment]
@jasonmit has begun working on the addon: https://github.com/jasonmit/legacy-bind-attr.

**Comment by rwjblue** [comment]
I left a few issues in that repo with some suggestions.

File: packages/ember-htmlbars/tests/compat/make_bound_helper_test.js [hunk 1]
```diff
@@ -101,7 +101,7 @@ QUnit.test('should update bound helpers in a subexpression when properties chang
   ignoreDeprecation(function() {
     view = EmberView.create({
       controller: { prop: 'isThing' },
-      template: compile('<div {{bind-attr data-foo=(dasherize prop)}}>{{prop}}</div>')
```
Comment (mixonic) [inline packages/ember-htmlbars/tests/compat/make_bound_helper_test.js]: This test is `should update bound helpers in a subexpression when properties change`, however here it has been changed to use an attribute expression and not a subexpression.

Can change this to still use a nested helper? It is specifically testing the subexpression hook.

File: packages/ember-htmlbars/tests/system/make_bound_helper_test.js [hunk 2]
```diff
@@ -39,7 +39,7 @@ QUnit.test('should update bound helpers in a subexpression when properties chang
     view = EmberView.create({
       container: container,
       controller: { prop: 'isThing' },
-      template: compile('<div {{bind-attr data-foo=(x-dasherize prop)}}>{{prop}}</div>')
```
Comment (mixonic) [inline packages/ember-htmlbars/tests/system/make_bound_helper_test.js]: This should also be testing a nested subexpression.

**Comment by mixonic** [comment]
This is awesome :-D :-D :-D IMO an addon was not required before dropping it (we only committed to addons for view and controller stuff) but it is great to have. :+1:

**Comment by rwjblue** [comment]
Agreed on the addon, but its presence seems good until @code0100fun and @abuiles get a way to auto rewrite templates for us :smiley:

**Comment by jasonmit** [comment]
Closed out the two issues and all tests are passing against this branch.  I'm ready to transfer ownership of the repository.  Thanks

File: packages/ember-htmlbars/tests/compat/make_bound_helper_test.js (hunk 1 above)
Comment (rwjblue) [inline packages/ember-htmlbars/tests/compat/make_bound_helper_test.js]: @mmun - ping

**Comment by rwjblue** [comment]
@mmun - This needs a few small touchup items before this can be merged...

File: packages/ember-htmlbars/tests/compat/make_bound_helper_test.js (hunk 1 above)
Comment (mmun) [inline packages/ember-htmlbars/tests/compat/make_bound_helper_test.js]: This is a nested helper: `(concat (dasherize prop))`.

Comment (mmun) [inline packages/ember-htmlbars/tests/compat/make_bound_helper_test.js]: I can make it more nested so that it's future-proof.

Comment (mixonic) [inline packages/ember-htmlbars/tests/compat/make_bound_helper_test.js]: @mmun I'm considering that putting a mustache in element space results in different hooks being called than putting it in the attr value. With helpers those differences were a constant pain, and I'm presuming this test is specifically for testing a binding in element space.

**Pull request merged**